packages/shell-api/src/field-level-encryption.ts

```diff
@@ -119,18 +108,11 @@ export class KeyVault extends ShellApiClass {
 
   @returnsPromise
   createKey(
-    kms: string | Document,
-    customMasterKey?: Document, // doc defined here: https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#masterkey
-    keyAltName?: string[]
+    kms: ClientEncryptionDataKeyProvider,
+    options?: ClientEncryptionCreateDataKeyProviderOptions
```

Comment:
https://docs.mongodb.com/manual/reference/method/KeyVault.createKey/ seems to want `keyAltName` as the third argument -- do we want to break from that?

Reply:
Yes, the documentation is invalid with the new specification of supporting GCP/Azure, too. I've updated the Technical Design document for that, too, after a small discussion we had: https://docs.google.com/document/d/1v0UUDvK4Ch8WM03XAFw1Ci-hem84vrgIw07RWCwmyJ4/edit?disco=AAAALAO7L-U. The API now matches the official spec and thus just forwards it to libmongocrypt.

Reply:
I’d still have a mild preference for backwards compatibility here as well, but as long as this is documented, I think it should be fine :)

Reply:
Actually, could we throw an exception when a third argument is passed then? I think that should make the upgrade path a lot easier than running the risk of silently ignoring options

Reply:
Yup sounds good, will add the check for additional args!

Reply:
Sorry, I just realized that this is not just affecting `keyAltNames`, but also `masterKey` … I would really suggest not doing this breakage here, if we don’t have a good reason to